src/components/phylogeny/Phylotree.vue

```diff
@@ -324,169 +359,213 @@ export default {
+                }
 
-          // Clear any existing menu items.
-          node.menu_items = [];
-
-          // Add a custom menu item to allow us to rename this node.
-          console.log("node", node);
-          addCustomMenu(
-            node,
-            (node) => "Rename this node",
-            () => {
-              const node = data;
-              const existingName = node.name || "(none)";
-              const newName = window.prompt(
-                `Rename node named '${existingName}' to:`
+                // Export the entire phylogeny as a Newick string, and store that
+                // in the phylogeny object.
+                const updatedNewickString = this.tree.getNewick();
+                console.log("updatedNewickString", updatedNewickString);
+                this.$store.commit("setPhylogenyProps", {
+                  phylogeny: this.phylogeny,
+                  newick: updatedNewickString,
+                });
+              },
+              (node) => true // We can replace this with a condition that indicates whether this node should be displayed.
+            );
+
+            // If the internal label has the same IRI as the currently selected
+            // phyloreference's reasoned node, further mark or label it as the resolved node.
+            //
+            // Note that this node might NOT be labeled, in which case we need to
+            // label it now!
+            if (
+              phyloref !== undefined &&
+              has(data, "@id") &&
+              this.$store.getters
+                .getResolvedNodesForPhylogeny(this.phylogeny, phyloref)
+                .includes(data["@id"])
+            ) {
+              // We found another pinning node!
+              if(!(data["@id"] in phylorefsByNodeId)) phylorefsByNodeId[data["@id"]] = new Set();
+              phylorefsByNodeId[data["@id"]].add(phyloref);
+              this.recurseNodes(data, (node) =>
+                pinningNodeChildrenIRIs.add(node["@id"])
               );
-              if (newName === null) {
-                // This means the user clicked "Cancel", so don't do anything.
-              } else if (!newName || newName === "undefined") {
-                // Apparently IE7 and IE8 will return the string 'undefined' if the user doesn't
-                // enter anything.
-                //
-                // Remove the current label.
-                node.name = "";
+
+              // Mark this node as the pinning node.
+              element.classed("pinning-node", true);
+
+              // If there is no circle, add one.
+              if (element.select("circle").empty()) {
+                element.append("circle").attr("cx", -3).attr("r", 4);
               } else {
-                // Set the new label.
-                node.name = newName;
+                // Make the pinning node circle larger (slightly larger than its usual size of 3).
+                element.select("circle").attr("r", 4);
```

Comment:
Perhaps the usual size should be a constant, and then we're just incrementing that constant here by an absolute (+1) or relative (+33%) value?